### PR TITLE
Omit child pages from view_feature by default

### PR DESCRIPTION
**Pull request by jwhitlock**
The feature hierarchy has two flavors - pages and rows.  Pages like [Web/API/KeyboardEvent](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent) are features, and the feature for ``Web/API`` is its parent.  Rows on the [compatibility table](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent#Browser_compatibility), like ``constructor`` are features, and the``Web/API/KeyboardEvent`` is its parent.

The problem comes when a page feature has both row children and page children, such as [Web/API/KeyboardEvent/key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key), which has ``Web/API/KeyboardEvent`` as a parent.

The default until now was for ``view_feature`` to return the entire descendant tree.  For ``Web/API/KeyboardEvent``, this includes row children like ``constructor`` but also page children like ``Web/API/KeyboardEvent/key``.  This means the API-generated table has more rows than the MDN table it is replacing.

This PR changes the default for ``view_feature`` to just the row children hierarchy, so that the pages are the same as the MDN pages.  The old behavior can be used by adding ``child_pages=1`` as a query param.

In the case of ``Web/API/KeyboardEvent``, this may actually be the desired behavior, to collect compatibility for sub-pages onto the parent pages.  This will require changes to the rendering code, and is out of scope for this PR.

### CODE REVIEW

File: webplatformcompat/cache.py [hunk 1]
```diff
@@ -176,11 +185,35 @@ def feature_v1_loader(self, pk):
...
+                hasattr(obj, '_row_children_pks') and
+                hasattr(obj, '_page_children_pks') and
+                hasattr(obj, '_row_descendant_pks')):
+            obj._children_pks = []
+            obj._row_children_pks = []
+            obj._page_children_pks = []
+            row_children = []
+            for child in obj.children.only('pk', 'mdn_uri'):
+                obj._children_pks.append(child.pk)
+                if child.mdn_uri:
```
Comment (groovecoder) [inline webplatformcompat/cache.py]: This is a bit confusing if you have just the code to look at. Could we add a comment here linking back to the docs that explain the difference between row children and page children?

File: webplatformcompat/tests/test_view_serializers.py [hunk 2]
```diff
@@ -349,33 +381,62 @@ def test_multiple_versions(self):
...
+        feature = self.setup_feature_tree()
+        url = reverse(
+            'viewfeatures-detail', kwargs={'pk': feature.pk, 'format': 'html'})
+        response = self.client.get(url, {'child_pages': True})
+        next_url = self.baseUrl + url + "?child_pages=1&page=2"
+        expected = '<a href="%s">next page</a>' % next_url
+        self.assertContains(response, expected, html=True)
+
+    @override_settings(PAGINATE_VIEW_FEATURE=4)
+    def test_just_right_feature_tree(self):
```
Comment (groovecoder) [inline webplatformcompat/tests/test_view_serializers.py]: lulz.

File: webplatformcompat/cache.py (hunk 1 above)
Comment (jwhitlock) [inline webplatformcompat/cache.py]: Yes, could use some explanation.  I don't think there's a good doc page to link to.  Maybe they should be model methods with a docstring, or just a big comment here.

Comment (groovecoder) [inline webplatformcompat/cache.py]: The [Including Child Pages](https://github.com/mdn/browsercompat/pull/65/files#diff-37ff62fc806144a51957c37e625c7dc4R80) doc section in this PR seems fine.

**Comment by groovecoder** [comment]
Just that nit. Then r+

**Comment by jwhitlock** [comment]
I've moved the commit from PR #70 to this one, and added the comment to the cache.

**Pull request merged**